ui/selectors/selectors.js

```diff
@@ -1262,7 +1262,7 @@ export function getKnownMethodData(state, data) {
   const fourBytePrefix = prefixedData.slice(0, 10);
   const { knownMethodData, use4ByteResolution } = state.metamask;
   // If 4byte setting is off, we do not want to return the knownMethodData
-  return use4ByteResolution ? knownMethodData?.[fourBytePrefix] : undefined;
+  return use4ByteResolution ? knownMethodData?.[fourBytePrefix] : {};
```

Comment:
for consistency, at the start of the method we could now return {} instead of null as well. 
```
  if (!data) {
    return null;
  }
```

```
const { name } = null; 
// VM334:1 Uncaught TypeError: Cannot destructure property 'name' of 'null' as it is null.
    at <anonymous>:1:9
```

Reply:
@digiwand since this is hotfix, I will prefer to keep scope minimum. More detailed changes can go in a followup PR.